Modules/_hashopenssl.c

```diff
@@ -88,22 +94,45 @@ typedef struct {
     PY_EVP_MD *evp_nosecurity;
 } py_hashentry_t;
 
+// Fundamental to TLS, assumed always present in any libcrypto:
 #define Py_hash_md5 "md5"
 #define Py_hash_sha1 "sha1"
 #define Py_hash_sha224 "sha224"
 #define Py_hash_sha256 "sha256"
 #define Py_hash_sha384 "sha384"
 #define Py_hash_sha512 "sha512"
-#define Py_hash_sha512_224 "sha512_224"
-#define Py_hash_sha512_256 "sha512_256"
-#define Py_hash_sha3_224 "sha3_224"
-#define Py_hash_sha3_256 "sha3_256"
-#define Py_hash_sha3_384 "sha3_384"
-#define Py_hash_sha3_512 "sha3_512"
-#define Py_hash_shake_128 "shake_128"
-#define Py_hash_shake_256 "shake_256"
-#define Py_hash_blake2s "blake2s"
-#define Py_hash_blake2b "blake2b"
+
+// Not all OpenSSL-like libcrypto libraries provide these:
+#if defined(NID_sha512_224)
+# define Py_hash_sha512_224 "sha512_224"
+#endif
+#if defined(NID_sha512_512)
+# define Py_hash_sha512_256 "sha512_256"
+#endif
```

Comment:
This typo would disable sha512-256 entirely, wouldn't it?
```suggestion
#if defined(NID_sha512_256)
# define Py_hash_sha512_256 "sha512_256"
#endif
```

Reply:
Thanks for the eyeballs, yes it would have. _(and since that isn't one of our guaranteed available hash functions it would've taken a while to notice)_

Reply:
or realistically, maybe it wouldn't have given which things tend to be defined at the same time.  still, correct change.